Dangerfile

```diff
@@ -81,9 +81,14 @@ end
 #   FUNCTIONS SECTION  #
 ########################
 
-def checkForNodeVersion(file, line)
+def checkForEnginesVersion(file)
   # Keep node version synced between declarations
-  warn("`package.json` node version was modified. Remember to update node version on `.travis.yml`, `.nvmrc` and `README`!") if file && line && file == "package.json" && line =~ /"node":/
+  fileDiff = git.diff_for_file(file)
+  nodeVersionMatches = fileDiff.patch.scan(/\+.*("node"|"npm"|"yarn")/)
+  nodeVersionMatches.each do |nodeVersionMatch|
+    stripMatch = nodeVersionMatch[0].gsub!('"', '`')
+    warn("#{stripMatch} version was modified in `package.json`. Remember to update #{stripMatch} version on `.travis.yml`, `.nvmrc` and `README`!") if file && nodeVersionMatch && file == "package.json"
```

Comment:
if you're already breaking into smaller functions, avoid using this inline if. Also, most of these validations should be done before scanning the whole file looking for node|npm|yarn.

Consider something along the lines of

```ruby
if file == "package.json"
  fileDiff = git.diff_for_file(file)
  engines = ["node","npm","yarn"]
  engines.each do |engine|

    nodeVersionMatches = fileDiff.patch.scan(/\+.*#{Regexp.quote(engine)}/)
      <get match, validate existence, fire warning>
    end
  end
end
```